### PR TITLE
Display userinfo

### PR DESCRIPTION
**Pull request by JohannesDoberer**

### CODE REVIEW

File: core/examples/luigi-sample-angular/src/luigi-config/extended/auth.js [hunk 1]
```diff
@@ -20,6 +20,14 @@ class Auth {
     authorizeMethod: 'GET',
     oAuthData: {
       client_id: 'egDuozijY5SVr0NSIowUP1dT6RVqHnlp'
+    },
+    userInfoFn: () => {
```
Comment (maxmarkus) [inline core/examples/luigi-sample-angular/src/luigi-config/extended/auth.js]: userInfoFn should give an input to the user. eg. the authorization data., how would you get the `Luigi User`?

Comment (maxmarkus) [reply core/examples/luigi-sample-angular/src/luigi-config/extended/auth.js]: Ah, if using mock  .. it is only using the provider's `userInfo`, but ignoring a provided `userInfoFn`. 
Maybe we change the implementation to allow to override the default provider userInfo function.

File: docs/authorization-configuration.md [hunk 2]
```diff
@@ -25,7 +25,8 @@ auth: {
     scope: 'audience:server:client_id:client openid profile email groups',
     redirect_uri: '',
     post_logout_redirect_uri: '/logout.html',
-    automaticSilentRenew: true
+    automaticSilentRenew: true,
+    userInfoFn:()=>{}
```
Comment (maxmarkus) [inline docs/authorization-configuration.md]: As a user, I'd like to get token info here, same the provider receives. `const tokenInfo = this.parseIdToken(authData.idToken);`  But watch out, OIDC provider does not have `getAuthData` function yet.
Might be enough to hand over `JSON.parse(localStorage.getItem('luigi.auth'))`.

File: docs/authorization-configuration.md [hunk 3]
```diff
@@ -74,6 +77,39 @@ auth: {
 - **response_type** defaults to the **id_token**. Any other parameter that is added to oAuthData is also added to the authorization payload.
 - **nonceFn** provides a function that returns a string in order to override the default **nonce**.
 - **logoutFn** provides the function to override the **logoutUrl** functionality for a custom logout. It needs to execute the **logoutCallback()** function after logout.
+- **userInfoFn** provides a function to get user information. It returns a promise of an user info object which contains user name and/or email to display it in the profile dropdown.
+
+
+### Custom Authentification Provider
```
Comment (bszwarc) [inline docs/authorization-configuration.md]: I'm wondering, should it be `authentification` or `authentication`? I was researching similar providers and it was `authentication` 🤔  but I might be wrong :) However, it is used to authenticate the user right?

File: docs/authorization-configuration.md [hunk 4]
```diff
@@ -74,6 +77,39 @@ auth: {
...
+
+If you are using any authentification provider you also have the possibility to implement some functions that luigi can deal with.
```
Comment (bszwarc) [inline docs/authorization-configuration.md]:
```suggestion
If you are using any authentication provider you can also implement some other functions for Luigi to use.
```

File: docs/authorization-configuration.md [hunk 5]
```diff
@@ -74,6 +77,39 @@ auth: {
...
+
+````
+export class CustomAuthentificationProvider {
+
+    login(){
+        // logic to handle the login mechanism
+        // returns a promise which contains a error message if something went wrong
```
Comment (bszwarc) [inline docs/authorization-configuration.md]:
```suggestion
        // returns a promise which contains an error message if something went wrong
```

File: docs/authorization-configuration.md (hunk 3 above)
Comment (JohannesDoberer) [reply docs/authorization-configuration.md]: You are totally right :)

File: docs/authorization-configuration.md [hunk 6]
```diff
@@ -74,6 +77,39 @@ auth: {
...
+    setTokenExpirationAction(){
+    }
+
+    generateNonce(){
+        //returns a string 
+    }
+
+    userInfo(){
+        // logic to get some user information
+        // returns a promise of a userinfo object which contains user name and/or email to display in the profile dropdown menu
```
Comment (bszwarc) [inline docs/authorization-configuration.md]:
```suggestion
        // returns a promise of a userinfo object which contains a user name and/or email to display in the profile dropdown menu
```

File: docs/authorization-configuration.md [hunk 7]
```diff
@@ -39,6 +40,7 @@ auth: {
 - **automaticSilentRenew** enables the automatic silent renewal of the token if it is supported by the server. The default value is `false`. For this mechanism to work, the browser must have third-party cookies support enabled.
 - **accessTokenExpiringNotificationTime** is the number of seconds before an access token is to expire and triggers silent token refresh. The default value is 60.
 - **thirdPartyCookiesScriptLocation** is the URL to the page containing third-party cookies support check. For details, see [Third-party cookies and silent token refresh section](#Third-party-cookies-and-silent-token-refresh).
+- **userInfoFn** provides a function to get user information. It returns a promise of a user info object which contains user name and/or email to display in the profile dropdown.
```
Comment (bszwarc) [inline docs/authorization-configuration.md]:
```suggestion
- **userInfoFn** provides a function to get user information. It returns a promise of a **userinfo** object which contains user name and/or email to display in the profile dropdown menu.
```

File: docs/authorization-configuration.md [hunk 8]
```diff
@@ -74,6 +77,39 @@ auth: {
 - **response_type** defaults to the **id_token**. Any other parameter that is added to oAuthData is also added to the authorization payload.
 - **nonceFn** provides a function that returns a string in order to override the default **nonce**.
 - **logoutFn** provides the function to override the **logoutUrl** functionality for a custom logout. It needs to execute the **logoutCallback()** function after logout.
+- **userInfoFn** provides a function to get user information. It returns a promise of a user info object which contains user name and/or email to display in the profile dropdown.
```
Comment (bszwarc) [inline docs/authorization-configuration.md]:
```suggestion
- **userInfoFn** provides a function to get user information. It returns a promise of a **userinfo** object which contains user name and/or email to display in the profile dropdown menu.
```

**Pull request merged**